runtime/pubsub/internal/gcp/push_handler.go

```diff
@@ -0,0 +1,74 @@
+package gcp
+
+import (
+	"encoding/json"
+	"net/http"
+	"strings"
+	"time"
+
+	"google.golang.org/api/idtoken"
+
+	"encore.dev/beta/errs"
+	"encore.dev/pubsub/internal/types"
+	"encore.dev/runtime"
+	"encore.dev/runtime/config"
+)
+
+// This is documented in https://cloud.google.com/pubsub/docs/push
+type pushPayload struct {
+	Message struct {
+		Attributes      map[string]string `json:"attributes"`
+		Data            []byte            `json:"data"`
+		MessageID       string            `json:"messageId"`
+		PublishTime     time.Time         `json:"publishTime"`
+		DeliveryAttempt int               `json:"deliveryAttempt,omitempty"` // Field documented in: https://cloud.google.com/pubsub/docs/handling-failures#track_delivery_attempts
+	} `json:"message"`
+	Subscription string `json:"subscription"`
+}
+
+func registerPushEndpoint(cfg *config.PubsubSubscription, f types.RawSubscriptionCallback) {
+	runtime.RegisterPubSubSubscriptionHandler(
+		cfg.ResourceID,
+		func(req *http.Request) error {
+			// If the request has not come from the Encore platform it must have
+			// a valid JWT set by Google.
+			if !runtime.IsEncoreAuthenticatedRequest(req.Context()) {
+				if err := validateGoogleJWT(req); err != nil {
```

Comment:
We need to validate the audience and service account email, otherwise this is easy to circumvent by just getting another JWT from Google for some other purpose

Reply:
@ekerfelt this means we need to add the service account to the Topic's config, so I can validate against it. ℹ️

Reply:
👍